src/Stratis.Bitcoin.Features.PoA.IntegrationTests/Tools/TestPoANetwork.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Collections.Generic;
+using NBitcoin;
+
+namespace Stratis.Bitcoin.Features.PoA.IntegrationTests.Tools
+{
+    public class TestPoANetwork : PoANetwork
```

Comment:
Come to think about it instead of a PoaNetwork you should have a PoaConsensusOptions that holds additional POA network stuff, the federated keys dont belong in network rather in consensus.
WDYT

Reply:
Fed keys are part of the network because we can have 2 networks which have exactly same consensus logic but different parameters of those that are network specific. 
So imho it's network specific.

Reply:
But we can move it to consensus options if needed because consensus option are in the network anyway. So I don't really mind

Reply:
yes agree its per network but the options are where you can use override for that purpose exactly (so you dont need to inherit network) please move to options if you dont mind. (you can do it in a separate PR)

Reply:
Ok, sure, will do